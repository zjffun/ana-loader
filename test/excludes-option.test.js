import { compile, getCompiler } from './helpers';

describe('"excludes" option', () => {
  it('should work with string array', async () => {
    const compiler = getCompiler('simple.js', {
      excludes: ['foo.tsx'],
    });
    const stats = await compile(compiler);
    const { modules } = stats.toJson();

    const result = modules.map((m) => {
      return {
        id: m.id,
        reasons: m.reasons.map((r) => r.moduleId),
      };
    });

    expect(result).toMatchSnapshot('result');
  });
});
