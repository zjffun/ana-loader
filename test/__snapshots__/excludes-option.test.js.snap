// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"excludes" option should work with string array: result 1`] = `
[
  {
    "id": "./simple.js",
    "reasons": [
      null,
    ],
  },
  {
    "id": "./foo.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.css",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.ts",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.vue",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.tsx",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.scss",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./foo.less",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./dynamic-import.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./commonjs.js",
    "reasons": [
      "./simple.js",
    ],
  },
  {
    "id": "./node_modules/test.js",
    "reasons": [
      "./foo.js",
      "./foo.ts",
      "./foo.vue",
    ],
  },
  {
    "id": "./fooDep.js",
    "reasons": [
      "./foo.js",
      "./foo.ts",
      "./foo.vue",
    ],
  },
  {
    "id": "./assets/test-font.woff",
    "reasons": [
      "./foo.css",
    ],
  },
  {
    "id": "./assets/test-font.ttf",
    "reasons": [
      "./foo.css",
    ],
  },
  {
    "id": "./assets/foo.png",
    "reasons": [
      "./foo.css",
      "./foo.less",
      "./foo.scss",
      "./foo.vue",
      "./foo.vue",
      "./foo.vue",
    ],
  },
  {
    "id": "./node_modules/testDep.js",
    "reasons": [
      "./node_modules/test.js",
    ],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
  {
    "id": "",
    "reasons": [],
  },
]
`;
